source/tool/chipsec/utilcmd/cmos_cmd.py

```diff
@@ -47,46 +40,52 @@ def cmos(argv):
     >>> chipsec_util cmos rl 0x0
     >>> chipsec_util cmos wh 0x0 0xCC
     """
-    if 3 > len(argv):
-        print cmos.__doc__
-        return
-
-    try:
-        _cmos = CMOS(  )
-    except CmosRuntimeError, msg:
-        print msg
-        return
-
-    op = argv[2]
-    t = time.time()
-
-    if ( 'dump' == op ):
-        logger().log( "[CHIPSEC] Dumping CMOS memory.." )
-        _cmos.dump()
-    elif ( 'readl' == op ):
-        off = int(argv[3],16)
-        val = _cmos.read_cmos_low( off )
-        logger().log( "[CHIPSEC] CMOS low byte 0x%X = 0x%X" % (off, val) )
-    elif ( 'writel' == op ):
-        off = int(argv[3],16)
-        val = int(argv[4],16)
-        logger().log( "[CHIPSEC] Writing CMOS low byte 0x%X <- 0x%X " % (off, val) )
-        _cmos.write_cmos_low( off, val )
-    elif ( 'readh' == op ):
-        off = int(argv[3],16)
-        val = _cmos.read_cmos_high( off )
-        logger().log( "[CHIPSEC] CMOS high byte 0x%X = 0x%X" % (off, val) )
-    elif ( 'writeh' == op ):
-        off = int(argv[3],16)
-        val = int(argv[4],16)
-        logger().log( "[CHIPSEC] Writing CMOS high byte 0x%X <- 0x%X " % (off, val) )
-        _cmos.write_cmos_high( off, val )
-    else:
-        logger().error( "unknown command-line option '%.32s'" % op )
-        print usage
-        return
-
-    logger().log( "[CHIPSEC] (cmos) time elapsed %.3f" % (time.time()-t) )
-
 
-chipsec_util.commands['cmos'] = {'func' : cmos,    'start_driver' : True, 'help' : cmos.__doc__  }
+    def requires_driver(self):
+        if len(self.argv) < 3:
+            return False
+        return True
+
+    def run(self):
+        if 3 > len(self.argv):

```

Comment:
While you're at it, can you please change this reverse logic to what you have above "len(self.argv) < 3".